### ISSUE TITLE
Avoid using Eigen::internal

### ISSUE THREAD

**Issue by fritzo** [nimble-dev/nimble#463]
Nimble's use of `Eigen::internal` was blocking upgrade to Eigen 3.3.4 (see our [eigen-3.3.4 branch](https://github.com/nimble-dev/nimble/tree/eigen-3.3.4)).

Current uses of `Eigen::internal`:
```sh
$ ag Eigen::internal nimble/inst/include/nimble/ nimble/inst/CppCode/
nimble/inst/include/nimble/nimbleEigen.h
30:  enum {nimbleUseLinearAccess = int(Eigen::internal::traits<T>::Flags & LinearAccessBit)};
721:  typedef typename Eigen::internal::traits<DerivedTarget>::Scalar Scalar;

nimble/inst/include/nimble/nimbleEigenNimArr.h
315:    nextBool = nimble_eigen_coeff_impl< Eigen::internal::traits<DerivedBool>::Flags & LinearAccessBit, int, DerivedBool, Eigen::Index >::getCoeff(BoolArg, i);
333:  typedef typename Eigen::internal::traits<DerivedX>::Scalar Scalar;
339:    thisTimes = nimble_eigen_coeff_impl< nimble_eigen_traits<DerivedTimes>::nimbleUseLinearAccess, typename Eigen::internal::traits<DerivedTimes>::Scalar, DerivedTimes, unsigned int >::getCoeff(EigenTimes, i);
```

**Comment by fritzo** [comment]
By version 3.3.4, Eigen has moved `Index` into the public namespace, so most of these usages should be resolved by #464 .

**Comment by fritzo** [comment]
Most of this was fixed in #464
